python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,380 @@
+"""Torch Module for PGExplainer"""
+import math
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.optim import Adam
+
+__all__ = ["PGExplainer"]
+
+
+class PGExplainer(nn.Module):
+    r"""PGExplainer from `Parameterized Explainer for Graph Neural Network
+    <https://arxiv.org/pdf/2011.04573>`
+
+    PGExplainer adopts a deep neural network (explanation network) to parameterize the generation
+    process of explanations, which enables it to explain multiple instances
+    collectively. PGExplainer models the underlying structure as edge
+    distributions, from which the explanatory graph is sampled.
+
+    Parameters
+    ----------
+    model : nn.Module
+        The GNN model to explain that tackles multiclass graph classification
+
+        * Its forward function must have the form
+          :attr:`forward(self, graph, nfeat, embed, edge_weight)`.
+        * The output of its forward function is the logits if embed=False else
+          return the intermediate node embeddings.
+    num_features : int
+        Node embedding size used by :attr:`model`.
+    epochs : int, optional
+        Number of epochs to train the explanation network. Default: 10.
+    lr : float, optional
+        Learning rate to train the explanation network. Default: 0.01.
+    coff_budget : float, optional
+        Size regularization to constrain the explanation size. Default: 0.01.
+    coff_connect : float, optional
+        Entropy regularization to constrain the connectivity of explanation. Default: 5e-4.
+    init_tmp : float, optional
+        The temperature at the first epoch. Default: 5.0.
+    final_tmp : float, optional
+        The temperature at the final epoch. Default: 1.0.
+    sample_bias : float, optional
+        Some members of a population are systematically more likely to be selected
+        in a sample than others. Default: 0.0.
+    """
+
+    def __init__(
+        self,
+        model,
+        num_features,
+        epochs=10,
+        learning_rate=0.01,
```

Comment:
```suggestion
        lr=0.01,
```

and all related logic